pkg/asset/machines/azure/machinesets.go

```diff
@@ -22,61 +22,119 @@ func MachineSets(clusterID string, config *types.InstallConfig, pool *types.Mach
 	}
 	platform := config.Platform.Azure
 	mpool := pool.Platform.Azure
+	azs := mpool.Zones
```

Comment:
if the length of azs is zero, setting the azs to `[]string{""}` should simplify a lot of the later calculations.
because no zones is equal to single "" zone..

Reply:
https://github.com/openshift/installer/pull/1949/commits/e66dd3fd0791eb5c901c82fc58ad630e86207976#diff-df3202277ffea159153362ec095a8762R180